src/App.tsx

```diff
@@ -15,47 +16,125 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
+enum SortType {
+  NONE = 'none',
+  ALPABET = 'alphabet',
+  LENGTH = 'length',
+}
+
+function getReorderedGoods(
+  goods: string[],
+  sortType: SortType,
+  isReversed: boolean,
+) {
+  const visibleGoods = [...goods];
+
+  if (sortType !== SortType.NONE) {
+    visibleGoods.sort((prevGood, currentGood) => {
+      switch (sortType) {
+        case SortType.ALPABET:
+          return prevGood.localeCompare(currentGood);
+
+        case SortType.LENGTH:
+          return prevGood.length - currentGood.length;
+
+        default:
+          return 0;
+      }
+    });
+  }
+
+  if (isReversed) {
+    visibleGoods.reverse();
+  }
+
+  return visibleGoods;
+}
+
 export const App: React.FC = () => {
+  const [isReversed, setIsReversed] = useState(false);
+  const [sortType, setSortType] = useState(SortType.NONE);
+
+  const handleReset = () => {
+    setIsReversed(false);
+    setSortType(SortType.NONE);
+  };
+
+  const visibleGoods = getReorderedGoods(
+    goodsFromServer,
+    sortType,
+    isReversed,
+  );
+
   return (
     <div className="section content">
       <div className="buttons">
         <button
           type="button"
-          className="button is-info is-light"
+          className={classNames(
+            'button',
+            'is-info',
+            { 'is-light': sortType !== SortType.ALPABET },
+          )}
+          onClick={() => setSortType(SortType.ALPABET)}
```

Comment:
It would be better to implement a handler instead of an inline function. 
Here some [info](https://www.anycodings.com/1questions/4778360/when-to-use-inline-function-on-button-onclick-event-javascriptreactjs)